modules/library/main/src/main/java/org/geotools/data/ows/URLCheckers.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2023, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.ows;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.geotools.util.factory.FactoryCreator;
+import org.geotools.util.factory.FactoryRegistry;
+import org.geotools.util.factory.GeoTools;
+
+/**
+ * Scans for all available URLChecker implementations, allows to register new ones, and provides a
+ * convenient method to apply them on a given location.
```

Comment:
In terms of conversion, URLs trades between files and URLs. This class does use File, it just adds some convenience method so that one can invoke it using URL and URI in addition to plain string. If we'd find the need to let it check a file, then yeah, I guess we'd go towards URL using URLs and then grab the external form.

Where do you see the overlap in the current code?

Reply:
URLs was a utility class for handling stuff with URL in a consistent manner; so URLCheckers may not of been needed to make an entire new class.